### ISSUE TITLE
[Bug] Directory permissions conflict, sensor publish feature

### ISSUE THREAD

**Issue by jos3neto** [awslabs/aws-iot-device-client#296]
Hello AWS IoT,

I'm trying to implement the sensor publish feature using an EC2 machine as a mock device. The OS is Ubuntu 22.04.

I'm getting a permissions conflict that I believe is a bug. The conflict is regarding the permissions to the directory that contains the unix socket server, in my case /tmp/sensors/. Here's how it looks:
<img width="361" alt="ls" src="https://user-images.githubusercontent.com/12755692/180246747-234be8a7-dac0-4da4-857c-95a4492e12c7.png">
"cpu" is the socket server file.

If the permissions to said directory is different from 660, the device client doesn't start and throws an error. Something like "desired: 660, actual: 700".

However, my "server.py" script needs execute access to the socket server, otherwise I get this:
<img width="460" alt="server" src="https://user-images.githubusercontent.com/12755692/180246906-4209c26b-09b9-4063-aefc-ca7d7bb7c790.png">
I know for a fact the problem is the "x" permission. If I change the permissions of /tmp/sensors/ to 700, server.py works perfectly. But then the device client doesn't start.

I'm thinking that permissions to /tmp/sensors/ the device client checks for should be 7XX, in line with every other directory permissions as per the image below:
<img width="725" alt="dirs" src="https://user-images.githubusercontent.com/12755692/180246985-3d05b6ad-1ac3-477f-a949-208f2dc93d0a.png">

Please advise and correct as needed.

Also please confirm that the permissions for the socket server file "cpu" are really 660.

Thanks!

**Comment by RogerZhongAWS** [comment]
Hello, thanks for reaching out! We actually ran into the same problem while testing and recently merged a PR that I believe should fix your issue: https://github.com/awslabs/aws-iot-device-client/pull/285

So to confirm, the "cpu" file should still be 660. The directory /tmp/sensors/ should be 700. Please let me know if it works after checking out the latest commit of device client.

Regards,
Roger

**Comment by jos3neto** [comment]
Hello Roger,

Thanks for the quick reply.
It seems we're not there yet, even with the latest commit checked out:

<img width="625" alt="Screen Shot 2022-07-21 at 19 59 22" src="https://user-images.githubusercontent.com/12755692/180329265-88915238-8d9c-47e5-a948-8b4cab735087.png">

Also please note that I changed the sensors directory to /home/ubuntu/dc/sensors/. I don't expect that to be a problem. Still getting the same error:
<img width="1229" alt="Screen Shot 2022-07-21 at 20 01 23" src="https://user-images.githubusercontent.com/12755692/180329312-f2f5bc66-6427-43c2-9a11-6a8dcce694cf.png">

**Comment by jos3neto** [comment]
Mistakenly closed, reopening.

**Comment by RogerZhongAWS** [comment]
@jos3neto Double check your config file and see if "addr" is the full filepath of your socket server file.

It should be something like this: `"addr": "/tmp/sensors/my-sensor-server-file"`
Not just the directory the contains the file, so not this: `"addr": "/tmp/sensors/"`

**Comment by jos3neto** [comment]
@RogerZhongAWS Here's the complete sensor publish config:

<img width="446" alt="Screen Shot 2022-07-22 at 08 25 38" src="https://user-images.githubusercontent.com/12755692/180429740-35ac1a9a-e354-4279-881b-8444c8a308e1.png">

**Comment by RogerZhongAWS** [comment]
I see, thanks for sharing that. We will look into this further and see if we can make this work with a python script.

Regards,
Roger

**Comment by RogerZhongAWS** [comment]
Hello, sorry for the late update.

After looking into this again, I still cannot see where device client is failing to validate these permissions correctly. There is only one place in the source code where we perform this check and AFAIK, we have separated the required permissions between the file and directory in #285. The documentation has also been updated to match.

I will keep testing to make sure, but so far I have not been able to reproduce this issue myself.
